src/sagas/tags.js

```diff
@@ -4,23 +4,39 @@ import Api from '../helpers/Api';
 import { actions, types } from '../redux/modules/tags';
 import { selectors as settings } from '../redux/modules/settings';
 import { actions as mdb } from '../redux/modules/mdb';
+import { selectors as filterSelectors } from '../redux/modules/filters';
+import { filtersTransformer } from '../filters';
+import { fetchViewsByUIDs } from './recommended';
 
 export function* fetchDashboard(action) {
-  const id = action.payload;
+  const { tag: id } = action.payload;
 
   try {
     const language = yield select(state => settings.getLanguage(state.settings));
-    const { data } = yield call(Api.tagDashboard, { id, language });
 
-    if (data && Array.isArray(data.latest_units)) {
-      yield put(mdb.receiveContentUnits(data.latest_units));
-      //  yield put(mdb.receiveContentUnits(data.promoted_units));
+    const filters      = yield select(state => filterSelectors.getFilters(state.filters, `topics_${id}`));
+    const filterParams = filtersTransformer.toApiParams(filters) || {};
 
-      yield put(actions.fetchDashboardSuccess(id, data));
-    } else {
-      const err = 'No latest units were found';
-      yield put(actions.fetchDashboardFailure(id, err));
-    }
+    const {
+      data: {
+        items,
+        media_total,
+        text_total
+      }
+    } = yield call(Api.tagDashboard, { ...action.payload, ...filterParams, language });
+
+    const cuIDs    = items.map(x => x.content_unit_id);
+    const labelIDs = items.map(x => x.label_id).filter(x => !!x);
+    const list     = items.map(({ content_unit_id, label_id, is_text }) => ({
+      cuID: content_unit_id,
+      lID: label_id,
+      isText: is_text
+    }));
+
+    yield put(mdb.fetchUnitsByIDs({ id: cuIDs }));
```

Comment:
Can you do all those in parallel?
Why one after another?
Are those done in parallel?

Reply:
done